csrc/ir/nodes.cpp

```diff
@@ -5211,6 +5215,71 @@ bool ForLoop::isGroup() const {
        typeid(kir::GroupedGridWelford)});
 }
 
+namespace {
+
+//! A utility class to check if runtime reduction exists
+class RuntimeReductionFinder : kir::ConstIrVisitor {
```

Comment:
Please have some assertion that the IR container is indeed KIR. `ForLoop` is no longer limited to KIR.

Reply:
added